tmt/steps/discover/fmf.py

```diff
@@ -147,6 +151,21 @@ def go(self):
         testdir = os.path.join(self.workdir, 'tests')
         dist_git_source = self.get('dist-git-source', False)
 
+        # Raise an exception if --fmf-id uses w/o --url and git root
+        # doesn't exist for ALL discovered plans
+        if self.opt('fmf_id') and \
+                not tmt.utils.Common.get_fmf_attr('discover', 'url'):
```

Comment:
url might be set in plan or CLI and `url = self.opt('url')` has already taken care getting the correct value
```suggestion
                not url:
```

Reply:
(same) This code is responsible for the case mentioned by you below https://github.com/teemtee/tmt/pull/932#discussion_r821672134

Reply:
There might be difference between fmf attributes and final value (cli preferences, future defaults etc.) IMO 'url' is already set correctly

Is there use case where `url` is different from `get_fmf_attr('discover', 'url')` and the later has expected value?

Reply:
With `non-url.fmf` and `url.fmf` in the same run.
```
(venv) [dkarpele@redhat plans]$ cat non-url.fmf 
summary: Basic smoke test
discover:
    how: fmf
execute:
    how: tmt
$ tmt run -rdddvvv discover -h fmf --fmf-id finish
``` 
If I stop debugger on this line then `url = None` and `tmt.utils.Common.get_fmf_attr('discover', 'url') = 'https://github.com/teemtee/tmt'`

But if we agree "that user should fix their command line to select plans which can success for 'discover --fmf-id'" then all this condition isn't necessary.

Reply:
But I need to know in advance that one of the plans doesn't have url before printing the output for working plan...

Reply:
Ah, I didn't see this use case

> But if we agree "that user should fix their command line to select plans which can success for 'discover --fmf-id'" then this condition isn't necessary.

Yes, please. Failing is IMO better than to hide tests which don't have url set.